### ISSUE TITLE
Number formatting isn't properly implemented

### ISSUE THREAD

**Issue by juliancoffee** [projectfluent/fluent-rs#269]
It seems that fluent-rs can parse options for numbers, but not use them.
Looking at FluentNumber::as_string(), the only option that gets used is minimum fraction digits.
```rust
    pub fn as_string(&self) -> Cow<'static, str> {
        let mut val = self.value.to_string();
        if let Some(minfd) = self.options.minimum_fraction_digits {
            if let Some(pos) = val.find('.') {
                let frac_num = val.len() - pos - 1;
                let missing = if frac_num > minfd {
                    0
                } else {
                    minfd - frac_num
                };
                val = format!("{}{}", val, "0".repeat(missing));
            } else {
                val = format!("{}.{}", val, "0".repeat(minfd));
            }
        }
        val.into()
    }
 ```
So, things to implement:
- [ ] Percent formatting
- [ ] Currency formatting
- [x] Decimal formatting (implicitly assumed, I guess)
- [ ] use_grouping
- [ ] minimum_integer_digits
- [x] minimum_fraction_digits
- [ ] maximum_fraction_digits
- [ ] minimum_significant_digits
- [ ] maximum_significant_digits

**Comment by juliancoffee** [comment]
@zbraniecki 
I thought about implementing percentage formatting, but I'm a bit worried about language-dependent formatting, do you have any advice on how to do it?

**Comment by zbraniecki** [comment]
I'd wait for ICU4X NumberFormatter, we have scaffolding for it in https://unicode-org.github.io/icu4x-docs/doc/fixed_decimal/index.html for some aspects, and https://github.com/unicode-org/icu4x/issues/1441 for others.

Maybe you'd like to help with that? :)

**Comment by juliancoffee** [comment]
> I'd wait for ICU4X NumberFormatter, we have scaffolding for it in https://unicode-org.github.io/icu4x-docs/doc/fixed_decimal/index.html for some aspects, and [unicode-org/icu4x#1441](https://github.com/unicode-org/icu4x/issues/1441) for others.
> 
> Maybe you'd like to help with that? :)

Yeah, I was looking at that a bit. What are the approximate steps for working on that?

**Comment by zbraniecki** [comment]
> What are the approximate steps for working on that?

You can go to issue #1441 and say you'd like to take it and ask for mentorship. Someone from our team will mentor you through that.

If you prefer some other feature of Number Formatter you can ask here https://github.com/unicode-org/icu4x/issues/275 and we'll help you carve out correct API to work on.

The concept in ICU4X is to divide what JS knows as `NumberFormat` into a tree of more specific and smaller formatters.

**Comment by gregtatum** [comment]
In Firefox we are using ICU4C to do this, but the examples in fluent-rs should be backed by ICU4X's number formatter. I'll update this issue to say as much.

**Comment by alerque** [comment]
I believe this work is already in progress and should be addressed as part of #335. Also of note #303 along the way.

**Comment by JasperDeSutter** [comment]
I'd be happy to open up #303 again, I can try and see if the approach taken for data providers in #335 works for number formatting as well.
